utils.groovy

```diff
@@ -199,4 +199,28 @@ def get_push_trigger() {
     ]
 }
 
+def change_metal_artifacts_list_order(artifacts) {
+    if (artifacts.contains('live')) {
+        artifacts.remove("metal")
+        artifacts.remove("metal4k")
+        artifacts.remove("live")
+
+        artifacts.add(0, "metal")
+        artifacts.add(1, "metal4k")
+        artifacts.add("live")
+    }
+    return artifacts
+}
+
+def get_artifacts_to_build(pipecfg, stream, basearch) {
+    def artifacts
+    if  (pipecfg.streams[stream].artifacts?."${basearch}" != null) {
+        artifacts = pipecfg.streams[stream]['artifacts'][basearch]
+    } else { // Merge default with additional artifacts
+        def default_artifacts =  pipecfg['default_artifacts'][basearch]
+        def additional_atifacts = pipecfg.streams[stream].additional_artifacts?."${basearch}"
+        artifacts = default_artifacts + additional_atifacts
+    }
+    return artifacts.unique()
```

Comment:
with this `unique` call here are we allowing users to mistakenly specify things twice and have it still work?

Reply:
Yeah exactly. I can imagine this happening when we promote an artifact to the default set.